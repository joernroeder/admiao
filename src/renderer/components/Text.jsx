import styled from '@emotion/styled'
import { space, typography, color } from 'styled-system'
import {
	verticalRhythm,
	verticalRhythmProps,
} from '../utils/styled-system-rhythm'

const Text = styled.p`
	${color};
	${space};
	${typography};
	${verticalRhythm};
	text-rendering: optimizeLegibility;
	-webkit-font-smoothing: antialiased;
	font-variant-ligatures: common-ligatures;
`

Text.defaultProps = {
	color: 'black',
	mt: 0,
	mb: 0,
	...verticalRhythmProps(0),
}

export default Text
