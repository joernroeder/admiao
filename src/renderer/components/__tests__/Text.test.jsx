import React from 'react'

import { render } from '@testing-library/react'

import Text from '../Text'

describe('Text component', () => {
	it('should render correctly', () => {
		const { container } = render(<Text />)
		const text = container.firstChild

		expect(text).toMatchSnapshot()
	})
})
