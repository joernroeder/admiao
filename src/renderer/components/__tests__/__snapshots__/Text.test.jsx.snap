// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should render correctly 1`] = `
.emotion-0 {
  color: black;
  margin-top: 0;
  margin-bottom: 0;
  font-size: 12px;
  line-height: 0;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

<p
  class="emotion-0"
  color="black"
  font-size="0"
/>
`;
